components/editor/nftEditorAdvanced.tsx

```diff
@@ -1,15 +1,68 @@
-import { Nft } from "@metaplex-foundation/js";
+import {Nft} from "@metaplex-foundation/js";
+import dynamic from "next/dynamic";
 
 type NftEditorAdvancedProps = {
-  nft: Nft;
+    nft: Nft;
 };
 
-const NftEditorAdvanced = ({ nft }: NftEditorAdvancedProps) => {
-  return (
-    <div>
-      <h1 className="text-xl font-bold">Advanced Editor</h1>
-    </div>
-  )
+const Ace = dynamic(
+    async () => {
+        const ace = await import('react-ace');
+        require('ace-builds/src-noconflict/mode-json');
+        require("ace-builds/src-noconflict/snippets/json");
+        require('ace-builds/src-noconflict/theme-monokai');
+        return ace;
+    },
+    {
+        loading: () => (
+            <p className="h-24 bg-back rounded-3xl p-2">Loading...</p>
+        ),
+        ssr: false,
+    })
+
+const NftEditorAdvanced = ({nft}: NftEditorAdvancedProps) => {
+    const nftString = JSON.stringify(nft, null, '\t')
```

Comment:
I think instead of this, you'll have to fetch the raw metadata file in a `useEffect` hook for example; the URI can be found in `nft.uri`.

Reply:
@abdelhakeem I thought the metadata would be fetched in the editor component and passed as params to the advanced and simple editor right?

Reply:
> @abdelhakeem I thought the metadata would be fetched in the editor component and passed as params to the advanced and simple editor right?

We actually pass down the `Nft` object itself, which has the parsed metadata for the simple editor under the `json` property, but for the advanced editor we'd need to fetch the raw file. The reason we pass down the `Nft` object as a whole is that we need it for the NFT update calls through Metaplex. Should we fetch both in the editor component and pass down each of them to its corresponding editor?

Reply:
I see thanks for the explanation! I think we should fetch it in the advanced editor since we're only using it in that component, I'll work on it and let you know